core/domain/story_services.py

```diff
@@ -846,8 +847,6 @@ def assign_role(committer, assignee, new_role, story_id):
     elif new_role == story_domain.ROLE_NONE:
         if story_rights.is_manager(assignee.user_id):
             story_rights.manager_ids.remove(assignee.user_id)
-        else:
```

Comment:
In an earlier comment, I asked:

> This doesn't make sense, I think -- why would we be changing NONE to NONE? Seems like an error case...
>
> (Make sure you have tests for this case.)

I don't see any tests? Tests are really important to prevent regressions in the future, so please make sure the existing tests cover all the possible cases. If there are tests missing, please add them.